packages/tldraw/src/lib/utils/frames/frames.ts

```diff
@@ -0,0 +1,106 @@
+import { Editor, TLFrameShape, TLShapeId, TLShapePartial, Vec2d, compact } from '@tldraw/editor'
+
+/**
+ * Remove a frame.
+ *
+ * @param ids - Ids of the frames you wish to remove.
+ * @param editor - tlraw editor instance.
+ *
+ * @public
+ */
+export function removeFrame(ids: TLShapeId[], editor: Editor) {
+	const frames = compact(
+		ids
+			.map((id) => editor.getShape<TLFrameShape>(id))
+			.filter((f) => f && editor.isShapeOfType<TLFrameShape>(f, 'frame'))
+	)
+	if (!frames.length) return
+
+	const allChildren: TLShapeId[] = []
+	editor.batch(() => {
+		frames.map((frame) => {
+			const children = editor.getSortedChildIdsForParent(frame.id)
+			if (children.length) {
+				editor.reparentShapes(children, frame.parentId, frame.index)
+				allChildren.push(...children)
+			}
+		})
+		editor.setSelectedShapes(allChildren)
+		editor.deleteShapes(ids)
+	})
+}
+
+/**
+ * Fit a frame to its content.
+ *
+ * @param id - Id of the frame you wish to fit to content.
+ * @param editor - tlraw editor instance.
+ *
+ * @public
+ */
+export function fitFrameToContent(id: TLShapeId, editor: Editor) {
+	const frame = editor.getShape<TLFrameShape>(id)
+	if (!frame) return
+
+	const childIds = editor.getSortedChildIdsForParent(frame.id)
+	const children = compact(childIds.map((id) => editor.getShape(id)))
+	if (!children.length) return
+
+	const bounds = {
```

Comment:
Not sure if there's a better way to do this. Tried with `getShapeGeometry` and then get bounds from there. But when I used the the transform on top of those bounds I was getting a different result that with current logic.

Reply:
seems to be working! I think you could also do `bounds = Box2d.FromPoints(shapes.map(...get vertices).flat())`